x-pack/plugins/apm/public/plugin.ts

```diff
@@ -71,21 +66,45 @@ export class ApmPlugin implements Plugin<ApmPluginSetup, ApmPluginStart> {
     this.initializerContext = initializerContext;
   }
   public setup(core: CoreSetup, plugins: ApmPluginSetupDeps) {
-    createCallApmApi(core.http);
     const config = this.initializerContext.config.get();
     const pluginSetupDeps = plugins;
 
     pluginSetupDeps.home.environment.update({ apmUi: true });
     pluginSetupDeps.home.featureCatalogue.register(featureCatalogueEntry);
 
     if (plugins.observability) {
+      const getApmDataHelper = async () => {
+        const { fetchOverviewPageData, hasData } = await import(
+          './application/dynamicImports'
+        );
+
+        return { fetchOverviewPageData, hasData };
+      };
       plugins.observability.dashboard.register({
         appName: 'apm',
-        fetchData: fetchOverviewPageData,
-        hasData,
+        hasData: async () => {
+          const dataHelper = await getApmDataHelper();
+          return await dataHelper.hasData();
+        },
+        fetchData: async (params: FetchDataParams) => {
+          const dataHelper = await getApmDataHelper();
+          return await dataHelper.fetchOverviewPageData(params);
+        },
       });
     }
 
+    const setupLazyStuff = async (coreStart: CoreStart) => {
+      // render APM feedback link in global help menu
+      const {
+        setHelpExtension,
+        setReadonlyBadge,
+        createCallApmApi,
+      } = await import('./application/dynamicImports');
```

Comment:
I think we prefer snake_case. We need to fix it for APM files, it's ignored by eslint for now.

Reply:
i have remove the file and have simplified it, at least that's how i look at it now :D